components/nimbus/src/error.rs

```diff
@@ -107,4 +109,11 @@ impl<'a> From<jexl_eval::error::EvaluationError<'a>> for NimbusError {
     }
 }
 
+#[cfg(feature = "stateful")]
+impl From<VersionParsingError> for NimbusError {
+    fn from(eval_error: VersionParsingError) -> Self {
+        NimbusError::VersionParsingError(eval_error.to_string())
+    }
+}
```

Comment:
The NimbusError should wrap this error isntead of coercing it to a string. Remove this and edit the VersionParsingError variant above:

```rust
#[cfg(feature="stateful")]
#[error("Error parsing a string into a version {0}")]
VersionParsingError(#[from] VersionParsingError),
```

Reply:
Oh this merged. Whelp nevermind.

Reply:
https://github.com/mozilla/application-services/pull/6459

Reply:
Thanks for having another look! Sorry for being too fast. I have to merge a bunch of PRs to unblock myself for the current PR I am working on. I fixed it!

Reply:
Ah, now I remember why I added it like this. 

```
rror[E0220]: associated type `VersionParsingError` not found for `Self`
  --> /home/circleci/project/target/debug/build/nimbus-sdk-29671292901466d4/out/cirrus.uniffi.rs:75:5
   |
75 |     r#VersionParsingError {
   |     ^^^^^^^^^^^^^^^^^^^^^ help: `Self` has the following associated type: `FfiType`
```

I already had to make a few changes because of the feature flag. `compare_version` is now behind the `stateful` feature flag, which has a larger impact on the codebase. I wasn't sure if you want to keep `VersionParsingError` in the `currius.uniffi` file or not.